app/controllers/concerns/shared_inventory_fetch.rb

```diff
@@ -0,0 +1,17 @@
+module SharedInventoryFetch
+  extend ActiveSupport::Concern
```

Comment:
What does extending `ActiveSupport::Concern` do for us here? Shouldn't `SharedInventoryFetch` be a service object under `app/services` or `lib`?